druid/src/main/scala/akka/stream/alpakka/druid/internal/TranquilizerStage.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.druid.internal
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.stream.alpakka.druid.TranquilizerSettings
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import akka.stream.stage._
+import com.metamx.tranquility.druid.DruidBeams
+import com.metamx.tranquility.tranquilizer.{ MessageDroppedException, Tranquilizer }
+import com.twitter.util.{ Return, Throw }
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ Future, Promise }
+import scala.util.{ Failure, Success, Try }
+
+class TranquilizerStage[A](settings: TranquilizerSettings[A]) extends GraphStage[FlowShape[A, Future[A]]] {
+
+  private val in = Inlet[A]("messages")
+  private val out = Outlet[Future[A]]("result")
+
+  override val shape = FlowShape(in, out)
+
+  @scala.throws[Exception](classOf[Exception])
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
+
+    val druidSender: Tranquilizer[A] = DruidBeams
+      .fromConfig(settings.dataSourceConfig, settings.timestamper, settings.objectWriter)
+      .partitioner(settings.partitioner)
+      .buildTranquilizer(settings.dataSourceConfig.tranquilizerBuilder())
+
+    new GraphStageLogic(shape) with StageLogging {
```

Comment:
extend InHandler and OutHandler directly by the logic. It will simplify the code a little bit and avoid two class headers so better memory-wise.